commit 0304f503411ee6e52e44f8afd1ee63e748aa8beb
Author: Yang Bo <yangbo@linuxdeepin.com>
Date:   Sun Jan 3 21:09:27 2016 +0800

    net: usb: cdc_ether: add Dell DW5580 as a mobile broadband adapter
    
    (commit 0b88393cdf6b1322522849e61f7a3328f4fd3843 upstream.)
    
    BugLink: https://launchpad.net/bugs/1513847
    
    Since Dell DW5580 is a 3G modem, this patch adds the device as a
    mobile broadband adapter
    
    Signed-off-by: Daniele Palmas <dnlplm@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    (cherry picked from commit 0b88393cdf6b1322522849e61f7a3328f4fd3843)
    Signed-off-by: Wen-chien Jesse Sung <jesse.sung@canonical.com>
    Reviewed-By: AceLan Kao <acelan.kao@canonical.com>
    Acked-by: Stefan Bader <stefan.bader@canonical.com>
    Acked-by: Tim Gardner <tim.gardner@canonical.com>
    Signed-off-by: Luis Henriques <luis.henriques@canonical.com>
    Signed-off-by: Yang Bo <yangbo@linuxdeepin.com>

diff --git a/drivers/net/usb/cdc_ether.c b/drivers/net/usb/cdc_ether.c
index 35a2bff..5e92076 100644
--- a/drivers/net/usb/cdc_ether.c
+++ b/drivers/net/usb/cdc_ether.c
@@ -764,6 +764,11 @@ static const struct usb_device_id	products[] = {
 			USB_CDC_SUBCLASS_ETHERNET, USB_CDC_PROTO_NONE),
 	.driver_info = (kernel_ulong_t) &wwan_info,
 }, {
+	/* Dell DW5580 modules */
+	USB_DEVICE_AND_INTERFACE_INFO(DELL_VENDOR_ID, 0x81ba, USB_CLASS_COMM,
+			USB_CDC_SUBCLASS_ETHERNET, USB_CDC_PROTO_NONE),
+	.driver_info = (kernel_ulong_t)&wwan_info,
+}, {
 	USB_INTERFACE_INFO(USB_CLASS_COMM, USB_CDC_SUBCLASS_ETHERNET,
 			USB_CDC_PROTO_NONE),
 	.driver_info = (unsigned long) &cdc_info,
